apps/coordinator/src/components/Wallet/TransactionsTab/FeeBumping/context/FeeBumpContext.tsx

```diff
@@ -0,0 +1,360 @@
+
+import { FeeBumpStatus, FeeBumpResult } from "../types";
+import {
+  useAnalyzeTransaction,
+  useChangeOutputIndex,
+  useGetAvailableUtxos,
+  useGetGlobalXpubs,
+} from "./hooks";
+
+// =============================================================================
+// CONTEXT TYPE DEFINITION
+// =============================================================================
+
+interface FeeBumpContextType {
+  state: FeeBumpingState;
+  dispatch: React.Dispatch<FeeBumpActionTypes>;
+
+  // Core fee bumping operations (handled internally by context)
+  createAcceleratedRBF: (options?: {
+    changeAddress?: string;
+    feeRate?: number;
+  }) => Promise<FeeBumpResult>;
+  createCancelRBF: (options?: {
+    cancelAddress?: string;
+    feeRate?: number;
+  }) => Promise<FeeBumpResult>;
+
+  // Loading states
+  isCreatingRBF: boolean;
+}
+
+// =============================================================================
+// MAIN CONTEXT PROVIDER
+// =============================================================================
+
+const FeeBumpContext = createContext<FeeBumpContextType | undefined>(undefined);
+
+interface FeeBumpProviderProps {
+  children: ReactNode;
+}
+
+export function FeeBumpProvider({ children }: FeeBumpProviderProps) {
+  // Core state management - never shared with child components
+  const [state, dispatch] = useReducer(feeBumpingReducer, initialState);
+  const [isCreatingRBF, setIsCreatingRBF] = useState(false); // Local loading state for RBF operations
+  // External dependencies
+  const availableUtxos = useGetAvailableUtxos();
+
+  // Getting all needed wallet state
+  const { network, addressType, requiredSigners, totalSigners } =
+    useSelector(selectWalletConfig);
+  const defaultChangeAddress = useSelector(
+    (state: any) => state.wallet?.change?.nextNode?.multisig?.address,
+  );
+
+  const changeOutputIndex = useChangeOutputIndex(state.transaction);
+  const globalXpubs = useGetGlobalXpubs();
+
+  // Use the analysis hook to handle transaction analysis and update state accordingly
+  // this will refresh if underlying data changes (such as transaction, fee estimates, etc.)
+  useAnalyzeTransaction();
+
+  // =============================================================================
+  // CORE FEE BUMPING OPERATIONS
+  // =============================================================================
+
+  // RBF OPERATIONS
+
+  /**
+   * Creates an accelerated RBF transaction that keeps the same outputs but with a higher fee
+   *
+   * This function performs the following steps:
+   * 1. Extracts UTXOs from wallet state
+   * 2. Identifies the change output in the original transaction
+   * 3. Creates an accelerated RBF transaction with the same outputs but higher fee
+   * 4. Returns the base64-encoded PSBT
+   *
+   * @param options Transaction and fee rate details
+   * @returns Promise resolving to the base64-encoded PSBT
+   */
+  const createAcceleratedRBF = useCallback(
```

Comment:
I suspect we could probably move these out as well but for now I think this is fine as these two are the only complex functions being created in the context provider. 

Honestly at some point we can make these all custom hooks that react to changes in the fee selections and then have accelerated, canceled, and CPFP all just created asynchronously and returned from the state as desired. That's probably a premature optimization though.